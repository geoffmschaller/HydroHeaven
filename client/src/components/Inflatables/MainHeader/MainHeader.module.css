.mainHeader {
  width: 100%;
  height: 800px;
  overflow: hidden;
  position: relative; }
  .mainHeader .video {
    width: 100%;
    position: absolute;
    top: 0;
    left: 0; }
  .mainHeader .img {
    width: 100%;
    position: absolute;
    top: 0;
    left: 0; }
  .mainHeader .mediaOverlay {
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    background-color: rgba(21, 67, 96, 0.5);
    z-index: 2;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center; }
    .mainHeader .mediaOverlay .contentShifter {
      width: 600px;
      margin: 0 auto;
      text-align: center; }
      .mainHeader .mediaOverlay .contentShifter img {
        width: 100%; }
      .mainHeader .mediaOverlay .contentShifter p {
        text-align: center;
        color: white;
        font-size: 20px; }
      .mainHeader .mediaOverlay .contentShifter button {
        color: #f8c471;
        border: 2px solid #f8c471;
        padding: 10px 80px; }
        .mainHeader .mediaOverlay .contentShifter button:hover {
          color: white;
          background-color: #f8c471; }

@media (max-width: 400px) {
  .mainHeader {
    height: 90vh; }
    .mainHeader .video, .mainHeader .img {
      height: 90vh;
      width: auto; }
    .mainHeader .mediaOverlay {
      padding: 0 10px; }
      .mainHeader .mediaOverlay .contentShifter {
        width: 100%; } }

@media (min-width: 401px) and (max-width: 800px) {
  .mainHeader {
    height: 90vh; }
    .mainHeader .video, .mainHeader .img {
      height: 90vh;
      width: auto; } }

@media (min-width: 801px) and (max-width: 1500px) {
  .mainHeader {
    height: 90vh; }
    .mainHeader .video, .mainHeader .img {
      height: 90vh;
      width: auto; } }
