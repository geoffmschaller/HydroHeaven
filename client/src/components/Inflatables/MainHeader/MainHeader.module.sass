
.mainHeader
	width: 100%
	height: 800px
	overflow: hidden
	position: relative

	.video
		width: 100%
		position: absolute
		top: 0
		left: 0

	.img
		width: 100%
		position: absolute
		top: 0
		left: 0

	.mediaOverlay
		width: 100%
		height: 100%
		position: absolute
		top: 0
		left: 0
		background-color: rgba(21, 67, 96, .5)
		z-index: 2
		display: flex
		flex-direction: column
		justify-content: center

		.contentShifter
			width: 600px
			margin: 0 auto
			text-align: center

			img
				width: 100%

			p
				text-align: center
				color: white
				font-size: 20px

			button
				color: #f8c471
				border: 2px solid #f8c471
				padding: 10px 80px

				&:hover
					color: white
					background-color: #f8c471

@media (max-width: 400px)
	.mainHeader
		height: 90vh

		.video, .img
			height: 90vh
			width: auto

		.mediaOverlay
			padding: 0 10px

			.contentShifter
				width: 100%

@media(min-width: 401px) and (max-width: 800px)
	.mainHeader
		height: 90vh

		.video, .img
			height: 90vh
			width: auto

@media(min-width: 801px) and (max-width: 1500px)
	.mainHeader
		height: 90vh

		.video, .img
			height: 90vh
			width: auto