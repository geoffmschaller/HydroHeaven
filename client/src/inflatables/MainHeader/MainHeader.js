import React from 'react';
import styles from './MainHeader.module.sass';
import HEADER_VIDEO from '../../static/video/spa_jets.mp4';
import {Link} from "react-router-dom";

class MainHeader extends React.Component {

	render() {
		return (
			<div id={styles.mainHeader}>
				<div className={styles.videoHolder}>
					<video src={HEADER_VIDEO} loop autoPlay/>
					<div id={styles.videoOverlay}>
						<div id={styles.title}>Ultimate In:</div>
						<div id={styles.largeTitle}>Relaxation <br/> & Hydro Therapy</div>
						<div id={styles.buttonBar}>
							{
								/*
								<button className={"btn-purple"}><i className="far fa-play"/> View Video</button>
								 */
							}
							<Link to={"/spas-hot-tubs/"}>
								<button className={"btn-white-transparant"}>Get Started</button>
							</Link>
							<Link to={"/spas-hot-tubs/grid/any"}>
								<button className={"btn-white-transparant"}>View More</button>
							</Link>
						</div>
					</div>
				</div>
			</div>
		);
	}
}

export default MainHeader;