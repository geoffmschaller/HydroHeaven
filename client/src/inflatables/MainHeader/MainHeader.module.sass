@import "src/static/sass/config"

#mainHeader
	width: 100%

	.videoHolder
		width: 100%
		margin: 0 auto
		height: 500px
		overflow: hidden
		-webkit-box-shadow: 10px 10px 5px 0 rgba(0, 0, 0, 0.36)
		-moz-box-shadow: 10px 10px 5px 0 rgba(0, 0, 0, 0.36)
		box-shadow: 10px 10px 5px 0 rgba(0, 0, 0, 0.36)
		position: relative

		video
			width: 100%

		#videoOverlay
			position: absolute
			bottom: 25%
			left: 15%
			color: white

			#title
				font-size: 16px
				text-transform: uppercase

			#largeTitle
				font-size: 80px
				text-transform: uppercase
				font-family: $montserat
				font-weight: bold
				line-height: 70px

			#buttonBar
				width: 100%
				max-width: 600px
				margin-top: 15px

				button
					margin-right: 10px

@media (max-width: 600px)
	#mainHeader
		.videoHolder
			height: 80vh

			video
				width: auto
				height: 100vh

			#videoOverlay
				left: 5%

				#title
					font-size: 12px

				#largeTitle
					font-size: 40px
					line-height: 40px

				#buttonBar
					button
						margin-bottom: 10px

@media (min-width: 601px) and (max-width: 1050px)
	#mainHeader
		.videoHolder
			height: 80vh

			video
				width: auto
				height: 100vh

			#videoOverlay
				bottom: 25%
				left: 5%