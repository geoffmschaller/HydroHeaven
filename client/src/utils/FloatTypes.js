export const FLOAT_LEFT = 'left';
export const FLOAT_RIGHT = 'right';