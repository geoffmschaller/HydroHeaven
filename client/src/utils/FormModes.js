export const FORM_READY = 'form_ready';
export const FORM_SUBMITTING = 'form_submitting';
export const FORM_SUCCESS = "form_sucess";