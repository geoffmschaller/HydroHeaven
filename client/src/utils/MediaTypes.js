export const VIDEO_FILE = 'video_file';
export const IMAGE_FILE = 'image_file';