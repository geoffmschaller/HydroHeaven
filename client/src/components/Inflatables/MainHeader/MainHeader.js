import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { IMAGE_FILE, VIDEO_FILE } from '../../../utils/MediaTypes';
import styles from './MainHeader.module.css';
import MAIN_LOGO from '../../../static/images/main_logo_desktop_shaddow.png';

const MainHeader = (props) => (
	<div className={styles.mainHeader}>
		{
			props.type === VIDEO_FILE
				? <video className={styles.video} src={props.media} autoPlay loop />
				: <img className={styles.image} src={props.media} alt="" />
		}
		<div className={styles.mediaOverlay}>
			<div className={styles.contentShifter}>
				<img src={MAIN_LOGO} alt="" />
				<p>
					Hydro Heaven is your best choice for Hot tub and Spa care. We have proudly served the central coast for decades and gladly offer ourexpertise to you.
				</p>
				<Link to="/spas-hot-tubs">
					<button type="button">Get Started</button>
				</Link>
			</div>
		</div>
	</div>
);

MainHeader.propTypes = {
	type: PropTypes.oneOf([VIDEO_FILE, IMAGE_FILE]).isRequired,
	media: PropTypes.string.isRequired,
};

export default MainHeader;
