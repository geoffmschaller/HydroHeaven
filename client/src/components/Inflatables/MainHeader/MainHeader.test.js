import React from 'react';
import MainHeader from "./MainHeader";
import MAIN_HEADER_TEST_IMAGE from "../../../static/images/headers/swim_spa_display_header.png";
import {VIDEO_FILE} from "../../../utils/MediaTypes";

const wrapper = <MainHeader type={VIDEO_FILE} media={MAIN_HEADER_TEST_IMAGE}/>;


describe("Main Header Inflatable", () => {

	it("Renders Correctly", () => {
		expect(wrapper).toMatchSnapshot();
	});

});
