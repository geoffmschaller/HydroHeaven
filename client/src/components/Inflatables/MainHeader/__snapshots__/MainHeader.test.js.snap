// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Header Inflatable 1`] = `
<MainHeader
  media="swim_spa_display_header.png"
  type="video_file"
/>
`;
